test/unit/spec/PasswordComplexityUtil_spec.js

```diff
@@ -0,0 +1,288 @@
+          expect(this.complexity.isEnabled(1)).toBe(true);
+        });
+
+        it('0 is disabled', function () {
+          expect(this.complexity.isEnabled(0)).toBe(false);
+        });
+
+        describe('validation if set to 2 letters', function () {
+          beforeEach(function () {
+            this.doesComplexityMeet = _.partial(this.complexity.doesComplexityMeet, 2);
+          });
+
+          it('a password with 2 lowercase letters meets the complexity', function () {
+            expect(this.doesComplexityMeet('12abCD')).toBe(true);
+          });
+
+          it('a password with 1 lowercase letters does not meet the complexity', function () {
+            expect(this.doesComplexityMeet('12AaBCD')).toBe(false);
+          });
+
+          it('empty password does not meet the complexity', function () {
+            expect(this.doesComplexityMeet('')).toBe(false);
+          });
+        });
+      });
+
+      describe('minUpperCase', function () {
+        beforeEach(function () {
+          this.complexity = this.complexities['minUpperCase'];
+        });
+
+        it('1 is enabled', function () {
+          expect(this.complexity.isEnabled(1)).toBe(true);
+        });
+
+        it('0 is disabled', function () {
+          expect(this.complexity.isEnabled(0)).toBe(false);
+        });
+
+        describe('validation if set to 3 letters', function () {
+          beforeEach(function () {
+            this.doesComplexityMeet = _.partial(this.complexity.doesComplexityMeet, 3);
+          });
+
+          it('a password with 3 uppercase characters meets the complexity', function () {
+            expect(this.doesComplexityMeet('34eFGH')).toBe(true);
+          });
+
+          it('a password with 2 uppercase character does not meet the complexity', function () {
+            expect(this.doesComplexityMeet('34efGH')).toBe(false);
+          });
+
+          it('empty password does not meet the complexity', function () {
+            expect(this.doesComplexityMeet('')).toBe(false);
+          });
+        });
+      });
+
+      describe('minNumber', function () {
+        beforeEach(function () {
+          this.complexity = this.complexities['minNumber'];
+        });
+
+        it('1 is enabled', function () {
+          expect(this.complexity.isEnabled(1)).toBe(true);
+        });
+
+        it('0 is disabled', function () {
+          expect(this.complexity.isEnabled(0)).toBe(false);
+        });
+
+        describe('validation if set to 4 numbers', function () {
+          beforeEach(function () {
+            this.doesComplexityMeet = _.partial(this.complexity.doesComplexityMeet, 4);
+          });
+
+          it('a password with 4 numbers meets the complexity', function () {
+            expect(this.doesComplexityMeet('12abCD34')).toBe(true);
+          });
+
+          it('a password with 3 numbers does not meet the complexity', function () {
+            expect(this.doesComplexityMeet('12abCD 4')).toBe(false);
```

Comment:
should we allow spaces in the password? it should probably be the same string but without the space ```'12abCD4'```